// confetti/dist/resources.js
import { eventsFindAllOptionsSchema, eventsFindOptionsSchema, } from './schemas/event.js';
import { contactsFindOptionsSchema, contactsFindAllOptionsSchema, } from './schemas/contact.js';
import { ticketsFindAllOptionsSchema, ticketsFindOptionsSchema, } from './schemas/ticket.js';
import { paymentsFindAllOptionsSchema, paymentsFindOptionsSchema, } from './schemas/payment.js';
import { webhooksFindAllOptionsSchema, webhooksFindOptionsSchema, } from './schemas/webhook.js';
import { workspacesFindAllOptionsSchema, workspacesFindOptionsSchema, } from './schemas/workspace.js';
import { categoriesFindAllOptionsSchema, categoriesFindOptionsSchema, } from './schemas/category.js';
import { ticketBatchesFindAllOptionsSchema, ticketBatchesFindOptionsSchema, } from './schemas/ticket-batch.js';
import { baseOptionsSchema } from './schemas/resource-options.js';
import models from './models/index.js';
export const eventsResource = {
    findAll: (options = {}, adapter) => {
        const validatedOptions = eventsFindAllOptionsSchema.parse(options);
        return adapter.get({ path: models.event.path, type: models.event.endpoint, ...validatedOptions });
    },
    find: (id, options = {}, adapter) => {
        const validatedOptions = eventsFindOptionsSchema.parse(options);
        return adapter.get({ path: `${models.event.path}/${id}`, type: models.event.endpoint, ...validatedOptions });
    },
};
export const ticketsResource = {
    findAll: (options, adapter) => {
        const validatedOptions = ticketsFindAllOptionsSchema.parse(options);
        return adapter.get({ path: models.ticket.path, type: models.ticket.endpoint, ...validatedOptions });
    },
    find: (id, options = {}, adapter) => {
        const validatedOptions = ticketsFindOptionsSchema.parse(options);
        return adapter.get({
            path: `${models.ticket.path}/${id}`,
            type: models.ticket.endpoint,
            ...validatedOptions,
        });
    },
    create: (json, options = {}, adapter) => {
        const validatedOptions = baseOptionsSchema.parse(options);
        if (!models.ticket.operations.create)
            throw new Error('Ticket create operation not found');
        const validatedData = models.ticket.operations.create.schema.parse(json);
        return adapter.post({
            // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/consistent-type-assertions
            json: validatedData,
            path: models.ticket.path,
            type: models.ticket.endpoint,
            ...validatedOptions,
        });
    },
};
export const contactsResource = {
    findAll: (options = {}, adapter) => {
        const validatedOptions = contactsFindAllOptionsSchema.parse(options);
        return adapter.get({ path: models.contact.path, type: models.contact.endpoint, ...validatedOptions });
    },
    find: (id, options = {}, adapter) => {
        const validatedOptions = contactsFindOptionsSchema.parse(options);
        return adapter.get({
            path: `${models.contact.path}/${id}`,
            type: models.contact.endpoint,
            ...validatedOptions,
        });
    },
    create: (json, options = {}, adapter) => {
        const validatedOptions = baseOptionsSchema.parse(options);
        if (!models.contact.operations.create)
            throw new Error('Contact create operation not found');
        const validatedData = models.contact.operations.create.schema.parse(json);
        return adapter.post({
            // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/consistent-type-assertions
            json: validatedData,
            path: models.contact.path,
            type: models.contact.endpoint,
            ...validatedOptions,
        });
    },
};
export const paymentsResource = {
    findAll: (options, adapter) => {
        const validatedOptions = paymentsFindAllOptionsSchema.parse(options);
        return adapter.get({ path: models.payment.path, type: models.payment.endpoint, ...validatedOptions });
    },
    find: (id, options = {}, adapter) => {
        const validatedOptions = paymentsFindOptionsSchema.parse(options);
        return adapter.get({
            path: `${models.payment.path}/${id}`,
            type: models.payment.endpoint,
            ...validatedOptions,
        });
    },
};
export const webhooksResource = {
    findAll: (options = {}, adapter) => {
        const validatedOptions = webhooksFindAllOptionsSchema.parse(options);
        return adapter.get({ path: models.webhook.path, type: models.webhook.endpoint, ...validatedOptions });
    },
    find: (id, options = {}, adapter) => {
        const validatedOptions = webhooksFindOptionsSchema.parse(options);
        return adapter.get({
            path: `${models.webhook.path}/${id}`,
            type: models.webhook.endpoint,
            ...validatedOptions,
        });
    },
    create: (json, options = {}, adapter) => {
        const validatedOptions = baseOptionsSchema.parse(options);
        if (!models.webhook.operations.create)
            throw new Error('Webhook create operation not found');
        const validatedData = models.webhook.operations.create.schema.parse(json);
        return adapter.post({
            path: models.webhook.path,
            // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/consistent-type-assertions
            json: validatedData,
            type: models.webhook.endpoint,
            ...validatedOptions,
        });
    },
    delete: (id, options = {}, adapter) => {
        const validatedOptions = webhooksFindOptionsSchema.parse(options);
        return adapter.delete({
            path: `${models.webhook.path}/${id}`,
            type: models.webhook.endpoint,
            ...validatedOptions,
        });
    },
};
export const workspacesResource = {
    findAll: (options = {}, adapter) => {
        const validatedOptions = workspacesFindAllOptionsSchema.parse(options);
        return adapter.get({
            path: models.workspace.path,
            type: models.workspace.endpoint,
            ...validatedOptions,
        });
    },
    find: (id, options = {}, adapter) => {
        const validatedOptions = workspacesFindOptionsSchema.parse(options);
        return adapter.get({
            path: `${models.workspace.path}/${id}`,
            type: models.workspace.endpoint,
            ...validatedOptions,
        });
    },
};
export const categoriesResource = {
    findAll: (options = {}, adapter) => {
        const validatedOptions = categoriesFindAllOptionsSchema.parse(options);
        return adapter.get({ path: models.category.path, type: models.category.endpoint, ...validatedOptions });
    },
    find: (id, options = {}, adapter) => {
        const validatedOptions = categoriesFindOptionsSchema.parse(options);
        return adapter.get({
            path: `${models.category.path}/${id}`,
            type: models.category.endpoint,
            ...validatedOptions,
        });
    },
};
export const ticketBatchesResource = {
    findAll: (options, adapter) => {
        const validatedOptions = ticketBatchesFindAllOptionsSchema.parse(options);
        return adapter.get({
            path: models.ticketBatch.path,
            type: models.ticketBatch.endpoint,
            ...validatedOptions,
        });
    },
    find: (id, options = {}, adapter) => {
        const validatedOptions = ticketBatchesFindOptionsSchema.parse(options);
        return adapter.get({
            path: `${models.ticketBatch.path}/${id}`,
            type: models.ticketBatch.endpoint,
            ...validatedOptions,
        });
    },
};
