// confetti/dist/schemas/resource-options.js
import { z } from 'zod';
export const baseOptionsSchema = z.object({
    raw: z.boolean().optional(),
    apiKey: z.string().optional(),
    apiHost: z.string().optional(),
    apiProtocol: z.string().optional(),
});
export const findBaseOptionsSchema = baseOptionsSchema.extend({
    include: z.array(z.string()).optional(),
});
export const pageOptionsSchema = z.object({
    number: z.number().optional(),
    size: z.number().optional(),
    offset: z.number().optional(),
    limit: z.number().optional(),
});
export const baseFindAllOptionsSchema = findBaseOptionsSchema.extend({
    filter: z
        .record(z.string(), z.union([z.string(), z.number(), z.boolean(), z.array(z.string()), z.array(z.number()), z.date()]))
        .optional(),
    sort: z.string().optional(),
    page: pageOptionsSchema.optional(),
});
// Static method schemas (apiKey required)
export const staticBaseOptionsSchema = z.object({
    raw: z.boolean().optional(),
    apiKey: z.string(),
    apiHost: z.string().optional(),
    apiProtocol: z.string().optional(),
});
export const staticBaseFindOptionsSchema = staticBaseOptionsSchema.extend({
    include: z.array(z.string()).optional(),
});
export const staticBaseFindAllOptionsSchema = staticBaseFindOptionsSchema.extend({
    filter: z
        .record(z.string(), z.union([z.string(), z.number(), z.boolean(), z.array(z.string()), z.array(z.number()), z.date()]))
        .optional(),
    sort: z.string().optional(),
    page: pageOptionsSchema.optional(),
});
