// confetti/dist/models/payment.js
import loadSamples from '../utils/load-samples.js';
import { schemaToAttributes } from '../utils/schema-to-attributes.js';
import { PaymentSchema } from '../schemas/payment.js';
import { extractFiltersFromSchema, extractSortingFromSchema, extractIncludesFromSchema, } from '../utils/resource-options-to-model.js';
import { paymentsFindAllOptionsSchema } from '../schemas/payment.js';
export default function PaymentModel() {
    return {
        key: 'payment',
        endpoint: 'payments',
        path: 'payments',
        name: 'Payment',
        sample: loadSamples('payment'),
        sorting: extractSortingFromSchema(paymentsFindAllOptionsSchema),
        filters: extractFiltersFromSchema(paymentsFindAllOptionsSchema),
        includes: extractIncludesFromSchema(paymentsFindAllOptionsSchema),
        operations: {
            read: {
                schema: PaymentSchema,
                attributes: schemaToAttributes(PaymentSchema),
            },
        },
        webhooks: [
            {
                type: 'payment.paid',
                label: 'Paid',
                description: 'Triggers when a payment is paid.',
            },
            {
                type: 'payment.refunded',
                label: 'Refunded',
                description: 'Triggers when a payment is refunded.',
            },
        ],
    };
}
