// confetti/dist/presenters/contact.js
export default function ({ presenters, Presenter }) {
    class ContactPresenterClass extends Presenter {
        static type = 'contact';
        static plural = 'contacts';
        attributes(contact) {
            // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/consistent-type-assertions
            const contactData = contact;
            if (contactData.workspaceId) {
                contactData.workspace = {
                    id: contactData.workspaceId,
                };
                delete contactData.workspaceId;
            }
            if (contactData.categoryIds?.length) {
                contactData.categories = contactData.categoryIds.map((id) => ({
                    id,
                }));
                delete contactData.categoryIds;
            }
            const s = super.attributes?.(contactData) || contactData;
            // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
            return s;
        }
        relationships() {
            return {
                workspace: presenters.WorkspacePresenter,
                categories: presenters.CategoryPresenter,
            };
        }
    }
    return ContactPresenterClass;
}
