// confetti/dist/presenters/ticket.js
export default function ({ presenters, Presenter }) {
    class TicketPresenterClass extends Presenter {
        static type = 'ticket';
        static plural = 'tickets';
        attributes(ticket) {
            // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/consistent-type-assertions
            const ticketData = ticket;
            if (ticketData.eventId) {
                ticketData.event = {
                    id: ticketData.eventId,
                };
                delete ticketData.eventId;
            }
            if (ticketData.ticketBatchId) {
                ticketData.ticketBatch = {
                    id: ticketData.ticketBatchId,
                };
                delete ticketData.ticketBatchId;
            }
            if (ticketData.sendEmailConfirmation !== undefined) {
                ticketData.meta = {
                    sendEmailConfirmation: ticketData.sendEmailConfirmation,
                };
            }
            delete ticketData.sendEmailConfirmation;
            const s = super.attributes?.(ticketData) || ticketData;
            // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
            return s;
        }
        relationships() {
            return {
                event: presenters.EventPresenter,
                ticketBatch: presenters.TicketBatchPresenter,
                addons: presenters.AddonPresenter,
            };
        }
    }
    return TicketPresenterClass;
}
