// confetti/dist/presenters/webhook.js
export default function ({ presenters, Presenter }) {
    class WebhookPresenterClass extends Presenter {
        static type = 'webhook';
        static plural = 'webhooks';
        attributes(webhook) {
            // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/consistent-type-assertions
            const webhookData = webhook;
            if (webhookData.eventId) {
                webhookData.event = {
                    id: webhookData.eventId,
                };
                delete webhookData.eventId;
            }
            if (webhookData.workspaceId) {
                webhookData.workspace = {
                    id: webhookData.workspaceId,
                };
                delete webhookData.workspaceId;
            }
            const s = super.attributes?.(webhookData) || webhookData;
            // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
            return s;
        }
        relationships() {
            return {
                event: presenters.EventPresenter,
                workspace: presenters.WorkspacePresenter,
            };
        }
    }
    return WebhookPresenterClass;
}
